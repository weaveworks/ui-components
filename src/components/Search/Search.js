import React from 'react';
import styled from 'styled-components';
import { map, includes, without, isEmpty, last, noop } from 'lodash';
import PropTypes from 'prop-types';
import requiredIf from 'react-required-if';

import { copyPropTypes } from '../../utils/compose';
import Input from '../Input';
import Dropdown from '../Dropdown';

import Term from './_SearchTerm';

const TermsContainer = styled.ul`
  list-style: none;
  display: flex;
  margin: 0;
  padding: 0;
  flex-wrap: wrap;

  ${props => props.disabled && 'opacity: 0.75;'};
`;

const Icon = styled.i`
  padding: 0 6px 0 10px;
  ${props => props.disabled && `color: ${props.theme.colors.gray600};`};
`;

const SearchInput = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 3;
  width: 100%;
`;

const Styled = component => styled(component)`
  position: relative;
  display: flex;

  ${props =>
    props.disabled && `pointer-events: none;`} background-color: ${props =>
  props.theme.colors[props.disabled ? 'gray50' : 'white']};
  border: 1px solid ${props => props.theme.colors.purple100};
  border-radius: ${props => props.theme.borderRadius.soft};
  font-size: ${props => props.theme.fontSizes.small};
  align-items: center;

  div,
  input {
    border: 0;
  }

  ${Input}, ${Dropdown} {
    padding: 0;
    margin: 0;
  }

  ${Dropdown} {
    flex: 1;
    line-height: 36px;
    border-left: 1px solid ${props => props.theme.colors.gray200};

    .dropdown-popover {
      width: auto;
    }
  }

  ${Input} {
    flex: 2;
    width: 100%;

    input {
      padding: 0 8px;
      width: 100%;
    }

    input:focus {
      outline: none;
    }
  }
`;

class Search extends React.PureComponent {
  addSearchTerm = value => {
    let nextPinnedTerms = this.props.pinnedTerms;
    // only push unique values
    if (!includes(nextPinnedTerms, value)) {
      nextPinnedTerms = [...nextPinnedTerms, value];
      this.props.onPin(nextPinnedTerms);
    }
    this.props.onChange('', nextPinnedTerms);
  };

  removeSearchTerm = value => {
    const nextPinnedTerms = without(this.props.pinnedTerms, value);
    this.props.onChange(this.props.query, nextPinnedTerms);
  };

  handleInputKeyPress = ev => {
    if (ev.key === 'Enter' && this.props.query.length > 0) {
      ev.preventDefault();
      this.addSearchTerm(this.props.query);
    } else if (ev.key === 'Backspace' && this.props.query === '') {
      ev.preventDefault();
      const term = last(this.props.pinnedTerms);
      if (term) {
        // Allow the user to edit the text of the last term instead of removing the whole thing.
        this.removeSearchTerm(term);
      }
    }
  };

  handleInputChange = ev => {
    this.props.onChange(ev.target.value, this.props.pinnedTerms);
  };

  handleFilterChange = (ev, value) => {
    this.input.focus();
    this.props.onFilterSelect(value);
  };

  render() {
    const {
      className,
      filters,
      placeholder,
      query,
      pinnedTerms,
      disabled,
    } = this.props;

    return (
      <div className={className}>
        <Icon className="fa fa-search" disabled={disabled} />
        <SearchInput>
          <TermsContainer disabled={disabled}>
            {map(pinnedTerms, term => (
              <Term key={term} term={term} onRemove={this.removeSearchTerm} />
            ))}
          </TermsContainer>
          <Input
            hideValidationMessage
            onChange={this.handleInputChange}
            value={query}
            onKeyDown={this.handleInputKeyPress}
            onBlur={this.props.onBlur}
            onFocus={this.props.onFocus}
            inputRef={ref => {
              this.input = ref;
            }}
            placeholder={pinnedTerms.length === 0 ? placeholder : null}
            disabled={disabled}
          />
        </SearchInput>

        {!isEmpty(filters) && (
          <Dropdown
            items={filters}
            disabled={disabled}
            placeholder="Filters"
            onChange={this.handleFilterChange}
          />
        )}
      </div>
    );
  }
}

Search.propTypes = {
  /**
   * Disables the component if true
   */
  disabled: PropTypes.bool,
  /**
   * A list of selectable filters to be rendered in a `<Dropdown />`.
   * When an option is clicked, the `value` is added to the search terms.
   * This array will be passed directly to the `items` prop of the `<Dropdown />.
   * If omitted, no dropdown will be rendered.
   */
  filters: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.string,
    })
  ),
  onBlur: PropTypes.func,
  /**
   * Handler that runs when the text input changes.
   * Returns the text as first argument, and the list of pinned terms as the second.
   */
  onChange: PropTypes.func.isRequired,
  /**
   * Handler that runs when an item from the search filter dropdown is selected
   * Returns the selected filter value.
   */
  onFilterSelect: requiredIf(PropTypes.func, props => props.filters.length > 0),
  onFocus: PropTypes.func,
  /**
   * Handler that runs when a search is pinned or unpinned.
   * Returns an array of the currently pinned terms.
   */
  onPin: PropTypes.func,
  /**
   * The initial pinned terms of the search field.
   * Changes to this prop will be ignored after initial render.
   */
  pinnedTerms: PropTypes.arrayOf(PropTypes.string).isRequired,
  /**
   * Text that will be passed to the search input as the placeholder.
   */
  placeholder: PropTypes.string,
  /**
   * The initial value to use to populate the search text field.
   * Changes to this prop will be ignored after initial render.
   */
  query: PropTypes.string.isRequired,
};

Search.defaultProps = {
  disabled: false,
  filters: [],
  onBlur: noop,
  onFilterSelect: noop,
  onFocus: noop,
  onPin: noop,
  placeholder: 'search',
};

export default copyPropTypes(Search, Styled(Search));
