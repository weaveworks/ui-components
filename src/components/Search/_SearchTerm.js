import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { noop } from 'lodash';

import { copyPropTypes } from '../../utils/compose';

const Styled = component => styled(component)`
  background-color: ${props => props.theme.colors.blue200};
  border-radius: ${props => props.theme.borderRadius.soft};
  margin: 4px 0 4px 2px;
  padding: 6px;
  margin-left: 4px;
  display: flex;
  align-items: center;

  i {
    padding-left: 4px;
    cursor: pointer;
  }
`;

class SearchTerm extends React.PureComponent {
  handleRemove = () => {
    this.props.onRemove(this.props.term, this.props.label);
  };

  render() {
    const { className, term, label } = this.props;
    return (
      <li className={`${className} search-term`}>
        <div className="search-term-text">{label || term}</div>
        <i onClick={this.handleRemove} className="fa fa-times remove-term" />
      </li>
    );
  }
}

SearchTerm.propTypes = {
  /**
   * Display value that will be rendered.
   */
  label: PropTypes.string,
  /**
   * Handler that will run when a term is removed.
   * The `term` prop will be passed to the `onRemove` handler.
   */
  onRemove: PropTypes.func,
  /**
   * The internal identifier for a term.
   * If no `label` is supplied, `term` will be rendered.
   */
  term: PropTypes.string.isRequired,
};

SearchTerm.defaultProps = {
  label: '',
  onRemove: noop,
};

export default copyPropTypes(SearchTerm, Styled(SearchTerm));
