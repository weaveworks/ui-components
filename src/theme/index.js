import { colors, themeColorsAsScss } from './colors';
import { fontSizes, fontFamilies, themeFontsAsScss } from './fonts';
import { borderRadius, themeBorderRadiiAsScss } from './borders';
import { layers, themeLayersAsScss } from './layers';
import { spacing, themeSpacingsAsScss } from './spacings';
import { overlayIconSize, textColor, themeMiscVarsAsScss } from './misc';
import { boxShadow } from './box-shadows';
import { atoms } from './atoms';

export function themeVarsAsScss() {
  const themeVariables = []
    .concat(themeColorsAsScss())
    .concat(themeLayersAsScss())
    .concat(themeFontsAsScss())
    .concat(themeBorderRadiiAsScss())
    .concat(themeSpacingsAsScss())
    .concat(themeMiscVarsAsScss());
  return `${themeVariables.join('; ')};`;
}

const theme = {
  // Component-specific
  atoms,

  // Borders
  borderRadius,

  // Box shadows
  boxShadow,

  // Colors
  colors,

  // Fonts
  fontFamilies,
  fontSizes,

  // z-index layers
  layers,

  // Misc
  overlayIconSize,

  // Spacings
  spacing,
  textColor,
};

export default theme;
