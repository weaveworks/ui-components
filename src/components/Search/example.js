import React from 'react';
import styled from 'styled-components';
import { map, filter, includes, noop } from 'lodash';

import { Example } from '../../utils/example';

import Search from './Search';

const List = ({ items, text, terms }) => {
  const list =
    text.length > 0 || terms.length > 0
      ? filter(items, i => text === i || includes(terms, i))
      : items;

  return <ul>{map(list, item => <li key={item}>{item}</li>)}</ul>;
};

const Styled = component => styled(component)`
  ${Example} {
    margin-bottom: 96px;
  }
`;

const NarrowSearch = styled(Search)`
  width: 300px;
`;

class SearchExample extends React.Component {
  state = {
    queries: {
      normal: { text: '', terms: [] },
      filters: { text: '', terms: [] },
    },
    query1: '',
    query2: '',
    query3: '',
    pinnedTerms1: [],
    pinnedTerms2: [],
    pinnedTerms3: ['foo:bar', 'is:automated'],
  };

  render() {
    return (
      <div className={this.props.className}>
        <Example>
          <h3>Normal</h3>
          <Search
            query={this.state.query1}
            pinnedTerms={this.state.pinnedTerms1}
            onChange={(query1, pinnedTerms1) =>
              this.setState({
                query1,
                pinnedTerms1,
                queries: {
                  ...this.state.queries,
                  normal: { text: query1, terms: pinnedTerms1 },
                },
              })
            }
          />
          <List
            items={['red', 'green', 'blue']}
            terms={this.state.queries.normal.terms}
            text={this.state.queries.normal.text}
          />
        </Example>
        <Example>
          <h3>With filters</h3>
          <NarrowSearch
            query={this.state.query2}
            pinnedTerms={this.state.pinnedTerms2}
            onFilterSelect={value =>
              this.setState({
                pinnedTerms2: [value],
                queries: {
                  ...this.state.queries,
                  filters: { text: '', terms: [value] },
                },
              })
            }
            onChange={(query2, pinnedTerms2) =>
              this.setState({
                query2,
                pinnedTerms2,
                queries: {
                  ...this.state.queries,
                  filters: { text: query2, terms: pinnedTerms2 },
                },
              })
            }
            filters={[
              { value: 'red', label: 'Red' },
              {
                value: 'blue',
                label: 'Blublublublublublublublublublublu',
              },
            ]}
          />
          <List
            items={['red', 'green', 'blue']}
            text={this.state.queries.filters.text}
            terms={this.state.queries.filters.terms}
          />
        </Example>
        <Example>
          <h3>With an initial query and filters</h3>
          <Search
            query={this.state.query3}
            pinnedTerms={this.state.pinnedTerms3}
            onChange={(query3, pinnedTerms3) =>
              this.setState({ query3, pinnedTerms3 })
            }
            onFilterSelect={value => this.setState({ pinnedTerms3: [value] })}
            filters={[
              { value: 'is:automated', label: 'Automated' },
              { value: 'is:locked', label: 'Locked' },
            ]}
          />
        </Example>
        <Example>
          <h3>Disabled</h3>
          <Search
            disabled
            query=""
            pinnedTerms={[]}
            filters={[{ value: 'red', label: 'Red' }]}
            onChange={noop}
          />
        </Example>
        <Example>
          <h3>Disabled with a query and pinnedTerms</h3>
          <Search
            disabled
            query="prom"
            pinnedTerms={this.state.pinnedTerms3}
            onChange={noop}
          />
        </Example>
      </div>
    );
  }
}

export default Styled(SearchExample);
