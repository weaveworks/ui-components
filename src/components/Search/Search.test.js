import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { createSpy } from 'expect';
import { noop } from 'lodash';

import { withTheme } from '../../utils/theme';

import Search from './Search';

describe('<Search />', () => {
  describe('snapshots', () => {
    it('renders empty', () => {
      const tree = renderer
        .create(
          withTheme(
            <Search
              query=""
              pinnedTerms={[]}
              onChange={noop}
              onFilterSelect={noop}
            />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a search string', () => {
      const tree = renderer
        .create(
          withTheme(
            <Search
              query="ui-server"
              pinnedTerms={[]}
              onChange={noop}
              onFilterSelect={noop}
            />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a group of terms', () => {
      const tree = renderer
        .create(
          withTheme(
            <Search
              query=""
              pinnedTerms={['deployments', 'default']}
              onChange={noop}
              onFilterSelect={noop}
            />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders filters', () => {
      const tree = renderer
        .create(
          withTheme(
            <Search
              query=""
              pinnedTerms={[]}
              onChange={noop}
              onFilterSelect={noop}
              filters={[
                { label: 'Automated', value: 'automated' },
                { label: 'Locked', value: 'locked' },
              ]}
            />
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('returns search terms', () => {
    const spy = createSpy();
    const search = mount(
      withTheme(
        <Search
          query=""
          pinnedTerms={['a']}
          onChange={spy}
          onFilterSelect={noop}
        />
      )
    );
    const $input = search.find('input');
    $input.simulate('change', { target: { value: 'myterm' } });
    expect(spy.calls[0].arguments).toEqual(['myterm', ['a']]);
  });

  it('calls the onFilterSelect prop with selected filter value', () => {
    const spy = createSpy();
    const search = mount(
      withTheme(
        <Search
          query=""
          pinnedTerms={[]}
          onChange={noop}
          onFilterSelect={spy}
          filters={[
            { label: 'Automated', value: 'automated' },
            { label: 'Locked', value: 'locked' },
          ]}
        />
      )
    );

    // Open the dialog pop up
    // TODO: make the dropdown do `display: none` instead of removing it from the dom?
    search
      .find('.dropdown-toggle')
      .first()
      .simulate('click');
    // select a filter
    search
      .find('.dropdown-item')
      .first()
      .simulate('click');
    expect(spy.calls[0].arguments).toEqual(['automated']);
  });

  it('clears terms', () => {
    const spy = createSpy();
    const search = mount(
      withTheme(
        <Search
          query=""
          pinnedTerms={['a', 'b']}
          onChange={spy}
          onFilterSelect={noop}
        />
      )
    );
    // Remove the 'a' search term
    search
      .find('.remove-term')
      .first()
      .simulate('click');
    expect(spy.calls[0].arguments).toEqual(['', ['b']]);
  });
});
