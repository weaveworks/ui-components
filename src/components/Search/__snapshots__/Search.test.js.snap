// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> snapshots renders a group of terms 1`] = `
.c8 {
  position: absolute;
  right: 10px;
  visibility: hidden;
  color: hsl(13,100%,40%);
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 36px;
}

.c7 input {
  padding-right: 12px;
  padding-left: 12px;
  width: 100%;
  line-height: 36px;
  box-shadow: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(0,0%,45%);
  border-radius: 2px;
  height: 36px;
  box-sizing: border-box;
}

.c7 input::-webkit-input-placeholder {
  color: hsl(0,0%,45%);
}

.c7 input::-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c7 input:-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c7 input:-ms-input-placeholder {
  color: hsl(0,0%,45%);
}

.c7 input::-webkit-input-placeholder {
  opacity: 1;
}

.c7 input::-moz-placeholder {
  opacity: 1;
}

.c7 input:-moz-placeholder {
  opacity: 1;
}

.c7 input:-ms-input-placeholder {
  opacity: 1;
}

.c7 input:disabled {
  background-color: hsl(0,0%,96%);
}

.c7 input:disabled::-webkit-input-placeholder {
  opacity: 0.5;
}

.c7 input:disabled::-moz-placeholder {
  opacity: 0.5;
}

.c7 input:disabled:-moz-placeholder {
  opacity: 0.5;
}

.c7 input:disabled:-ms-input-placeholder {
  opacity: 0.5;
}

.c9 {
  display: none;
  padding-left: 8px;
  margin-top: 8px;
  font-size: 14px;
  text-align: left;
  color: inherit;
  visibility: hidden;
}

.c6 {
  padding: 8px;
}

.c4 {
  background-color: hsl(191,100%,80%);
  border-radius: 2px;
  margin: 4px 0 4px 2px;
  padding: 6px;
  margin-left: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 i {
  padding-left: 4px;
  cursor: pointer;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  padding: 0 6px 0 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 2px;
  font-size: 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c5,
.c0 .c10 {
  padding: 0;
  margin: 0;
}

.c0 .c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 36px;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c10 .dropdown-popover {
  width: auto;
}

.c0 .c5 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c5 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c5 input:focus {
  outline: none;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
    disabled={false}
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
      disabled={false}
    >
      <li
        className="c4 search-term"
      >
        <div
          className="search-term-text"
        >
          deployments
        </div>
        <i
          className="fa fa-times remove-term"
          onClick={[Function]}
        />
      </li>
      <li
        className="c4 search-term"
      >
        <div
          className="search-term-text"
        >
          default
        </div>
        <i
          className="fa fa-times remove-term"
          onClick={[Function]}
        />
      </li>
    </ul>
    <div
      className="c5 c6"
    >
      
      <div
        className="c7"
      >
        <input
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder={null}
          value=""
        />
        <i
          className="fa fa-times-circle c8"
        />
      </div>
      <span
        className="c9"
      >
        
      </span>
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders a search string 1`] = `
.c7 {
  position: absolute;
  right: 10px;
  visibility: hidden;
  color: hsl(13,100%,40%);
}

.c6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 36px;
}

.c6 input {
  padding-right: 12px;
  padding-left: 12px;
  width: 100%;
  line-height: 36px;
  box-shadow: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(0,0%,45%);
  border-radius: 2px;
  height: 36px;
  box-sizing: border-box;
}

.c6 input::-webkit-input-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input::-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input:-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input:-ms-input-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input::-webkit-input-placeholder {
  opacity: 1;
}

.c6 input::-moz-placeholder {
  opacity: 1;
}

.c6 input:-moz-placeholder {
  opacity: 1;
}

.c6 input:-ms-input-placeholder {
  opacity: 1;
}

.c6 input:disabled {
  background-color: hsl(0,0%,96%);
}

.c6 input:disabled::-webkit-input-placeholder {
  opacity: 0.5;
}

.c6 input:disabled::-moz-placeholder {
  opacity: 0.5;
}

.c6 input:disabled:-moz-placeholder {
  opacity: 0.5;
}

.c6 input:disabled:-ms-input-placeholder {
  opacity: 0.5;
}

.c8 {
  display: none;
  padding-left: 8px;
  margin-top: 8px;
  font-size: 14px;
  text-align: left;
  color: inherit;
  visibility: hidden;
}

.c5 {
  padding: 8px;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  padding: 0 6px 0 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 2px;
  font-size: 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c4,
.c0 .c9 {
  padding: 0;
  margin: 0;
}

.c0 .c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 36px;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c9 .dropdown-popover {
  width: auto;
}

.c0 .c4 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c4 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c4 input:focus {
  outline: none;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
    disabled={false}
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
      disabled={false}
    />
    <div
      className="c4 c5"
    >
      
      <div
        className="c6"
      >
        <input
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="search"
          value="ui-server"
        />
        <i
          className="fa fa-times-circle c7"
        />
      </div>
      <span
        className="c8"
      >
        
      </span>
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders empty 1`] = `
.c7 {
  position: absolute;
  right: 10px;
  visibility: hidden;
  color: hsl(13,100%,40%);
}

.c6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 36px;
}

.c6 input {
  padding-right: 12px;
  padding-left: 12px;
  width: 100%;
  line-height: 36px;
  box-shadow: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(0,0%,45%);
  border-radius: 2px;
  height: 36px;
  box-sizing: border-box;
}

.c6 input::-webkit-input-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input::-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input:-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input:-ms-input-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input::-webkit-input-placeholder {
  opacity: 1;
}

.c6 input::-moz-placeholder {
  opacity: 1;
}

.c6 input:-moz-placeholder {
  opacity: 1;
}

.c6 input:-ms-input-placeholder {
  opacity: 1;
}

.c6 input:disabled {
  background-color: hsl(0,0%,96%);
}

.c6 input:disabled::-webkit-input-placeholder {
  opacity: 0.5;
}

.c6 input:disabled::-moz-placeholder {
  opacity: 0.5;
}

.c6 input:disabled:-moz-placeholder {
  opacity: 0.5;
}

.c6 input:disabled:-ms-input-placeholder {
  opacity: 0.5;
}

.c8 {
  display: none;
  padding-left: 8px;
  margin-top: 8px;
  font-size: 14px;
  text-align: left;
  color: inherit;
  visibility: hidden;
}

.c5 {
  padding: 8px;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  padding: 0 6px 0 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 2px;
  font-size: 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c4,
.c0 .c9 {
  padding: 0;
  margin: 0;
}

.c0 .c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 36px;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c9 .dropdown-popover {
  width: auto;
}

.c0 .c4 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c4 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c4 input:focus {
  outline: none;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
    disabled={false}
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
      disabled={false}
    />
    <div
      className="c4 c5"
    >
      
      <div
        className="c6"
      >
        <input
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="search"
          value=""
        />
        <i
          className="fa fa-times-circle c7"
        />
      </div>
      <span
        className="c8"
      >
        
      </span>
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders filters 1`] = `
.c7 {
  position: absolute;
  right: 10px;
  visibility: hidden;
  color: hsl(13,100%,40%);
}

.c6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 36px;
}

.c6 input {
  padding-right: 12px;
  padding-left: 12px;
  width: 100%;
  line-height: 36px;
  box-shadow: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(0,0%,45%);
  border-radius: 2px;
  height: 36px;
  box-sizing: border-box;
}

.c6 input::-webkit-input-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input::-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input:-moz-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input:-ms-input-placeholder {
  color: hsl(0,0%,45%);
}

.c6 input::-webkit-input-placeholder {
  opacity: 1;
}

.c6 input::-moz-placeholder {
  opacity: 1;
}

.c6 input:-moz-placeholder {
  opacity: 1;
}

.c6 input:-ms-input-placeholder {
  opacity: 1;
}

.c6 input:disabled {
  background-color: hsl(0,0%,96%);
}

.c6 input:disabled::-webkit-input-placeholder {
  opacity: 0.5;
}

.c6 input:disabled::-moz-placeholder {
  opacity: 0.5;
}

.c6 input:disabled:-moz-placeholder {
  opacity: 0.5;
}

.c6 input:disabled:-ms-input-placeholder {
  opacity: 0.5;
}

.c8 {
  display: none;
  padding-left: 8px;
  margin-top: 8px;
  font-size: 14px;
  text-align: left;
  color: inherit;
  visibility: hidden;
}

.c5 {
  padding: 8px;
}

.c13 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  padding-right: 28px;
  padding-left: 12px;
  cursor: pointer;
}

.c11 {
  height: 36px;
  box-sizing: border-box;
  border-radius: 2px;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(0,0%,45%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 .c12 {
  padding: 0;
}

.c11 div:last-child {
  margin-left: auto;
}

.c14 {
  position: absolute;
  right: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c10 {
  height: 36px;
  line-height: 34px;
  position: relative;
  width: 256px;
  box-sizing: border-box;
}

.c3 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  padding: 0 6px 0 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(240,20%,90%);
  border-radius: 2px;
  font-size: 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 div,
.c0 input {
  border: 0;
}

.c0 .c4,
.c0 .c9 {
  padding: 0;
  margin: 0;
}

.c0 .c9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 36px;
  border-left: 1px solid hsl(0,0%,80%);
}

.c0 .c9 .dropdown-popover {
  width: auto;
}

.c0 .c4 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
}

.c0 .c4 input {
  padding: 0 8px;
  width: 100%;
}

.c0 .c4 input:focus {
  outline: none;
}

<div
  className="c0"
>
  <i
    className="fa fa-search c1"
    disabled={false}
  />
  <div
    className="c2"
  >
    <ul
      className="c3"
      disabled={false}
    />
    <div
      className="c4 c5"
    >
      
      <div
        className="c6"
      >
        <input
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="search"
          value=""
        />
        <i
          className="fa fa-times-circle c7"
        />
      </div>
      <span
        className="c8"
      >
        
      </span>
    </div>
  </div>
  <div
    className="c9 c10"
    title="Filters"
  >
    <div
      className="dropdown-toggle c11 c12 c13"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="c12 c13"
        disabled={false}
      >
        Filters
      </div>
      <span
        className="fa fa-caret-down c14"
      />
    </div>
  </div>
</div>
`;
